microsoft/botframework-solutions
# [TypeScript][BotBuilder-Libs] Migrate from TSLint to ESLint

## Purpose
*What is the context of this pull request? Why is it being done?*

The current implementation of TSLint will be deprecated (https://github.com/palantir/tslint/issues/4534).
This pull-request migrates from TSLint to ESLint the following projects:
- botbuilder-solutions
- botbuilder-skills

## Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp?*

- Add `ESLint`
- Configure `ESLint` with the `TSLint` rules.
- Fix the new linting issues.

*Include illustrative screenshots for context if applicable.*
![image](https://user-images.githubusercontent.com/12394167/60470446-e07d8400-9c36-11e9-8bd2-dbf5fd028f5c.png)

## Tests
*Is this covered by existing tests or new ones? If no, why not?*

1. Go to `./lib/typescript` and open a PowerShell terminal.
1. Run `rush install`
1. Run `rush build`
1. Run `rush lint`
1. Run `rush test`
1. Check that the project doesn't have linting issues
